api/utils/health_utils.py

```
@@ -246,6 +246,21 @@ def check_minio_alive():
 
 
+def check_s3_alive():
+    """
+    Check AWS S3 (or any S3-compatible) liveness via the active
+    storage backend's `.health()` method. Delegates to the generic
+    ``check_storage`` so the same check works for AWS S3, MinIO,
+    R2, and any other S3-compatible endpoint. See #17294.
+    """
+    ok, payload = check_storage()
+    if ok:
+        logging.debug("check_s3_alive: ok, elapsed=%s ms", payload.get("elapsed", "?"))
```

Ruff (0.15.21)

[error] Undefined name `logging`

(F821)